### PR TITLE
remove decimals for Colombian (COP) and Argentina's Peso (ARS)

### PR DESCRIPTION
**Pull request by TChukwuleta**
Fix for issue #5708 

Remove decimal places for COP and ARS currencies

### CODE REVIEW

File: BTCPayServer/wwwroot/pos/common.js [hunk 1]
```diff
@@ -93,10 +93,16 @@ const posCommon = {
         formatCurrency (value, withSymbol) {
             const currency = this.currencyCode
             if (currency === 'BTC' || currency === 'SATS') return this.formatCrypto(value, withSymbol)
-            const { divisibility } = this.currencyInfo
+
+            // Check if the currency is COP or ARS (exclude decimal places)
+            const excludeDecimalPlaces = ['COP', 'ARS'].includes(currency);
```
Comment (Kukks) [inline BTCPayServer/wwwroot/pos/common.js:98]: This should not be hardcoded in the js but rather decides based on divisibility, which is set in a currency service in c#.

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/wwwroot/pos/common.js (hunk 1 above)
Comment (TChukwuleta) [reply BTCPayServer/wwwroot/pos/common.js:98]: Noted @Kukks. Would look into it

**Review by matman7x — APPROVED** [review]

**Comment by NicolasDorier** [comment]
why does `numberFormatInfo.CurrencyDecimalDigits` doesn't returns 0 for those two currencies?

**Comment by TChukwuleta** [comment]
@NicolasDorier 

numberFormatInfo (a method under system.Globalization) has default value for CurrencyDecimalDigits to be 2

I am not sure if  or how it is tied to currencies

**Comment by NicolasDorier** [comment]
@TChukwuleta , please instead of doing this hack here, add those currencies to https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer.Rating/Currencies.json

Then extend the test `RoundupCurrenciesCorrectly` to make sure COP and ARS have the right `CurrencyDecimalDigits`

File: BTCPayServer.Rating/CurrencyNameTable.cs [hunk 2]
```diff
@@ -77,7 +77,19 @@ public IFormatProvider GetCurrencyProvider(string currency)
                             continue;
                         try
                         {
-                            _CurrencyProviders.TryAdd(new RegionInfo(culture.LCID).ISOCurrencySymbol, culture);
+                            if (new RegionInfo(culture.LCID).ISOCurrencySymbol.Equals("ARS") || new RegionInfo(culture.LCID).ISOCurrencySymbol.Equals("COP"))
```
Comment (NicolasDorier) [inline BTCPayServer.Rating/CurrencyNameTable.cs:80]: Remove this, you shouldn't have anything hard coded.

File: BTCPayServer.Rating/Currencies.json [hunk 3]
```diff
@@ -58,7 +58,7 @@
    {
       "name":"Argentine Peso",
       "code":"ARS",
-      "divisibility":2,
```
Comment (NicolasDorier) [inline BTCPayServer.Rating/Currencies.json:61]: https://en.wikipedia.org/wiki/Argentine_peso it says there is 2 subunit.

File: BTCPayServer.Rating/Currencies.json [hunk 4]
```diff
@@ -289,7 +289,7 @@
    {
       "name":"Colombian Peso",
       "code":"COP",
-      "divisibility":2,
```
Comment (NicolasDorier) [inline BTCPayServer.Rating/Currencies.json:292]: https://en.wikipedia.org/wiki/Colombian_peso it says there is 2 subunit

File: BTCPayServer.Tests/FastTests.cs [hunk 5]
```diff
@@ -790,7 +790,7 @@ public void RoundupCurrenciesCorrectly()
                 (0.0005m, "0.0005 USD", "USD"), (0.001m, "0.001 USD", "USD"), (0.01m, "0.01 USD", "USD"),
                 (0.1m, "0.10 USD", "USD"), (0.1m, "0,10 EUR", "EUR"), (1000m, "1,000 JPY", "JPY"),
                 (1000.0001m, "1,000.00 INR", "INR"),
-                (0.0m, "0.00 USD", "USD")
```
Comment (NicolasDorier) [inline BTCPayServer.Tests/FastTests.cs:793]: Note that you should add a test on `CurrencyNameTable.Instance.GetNumberFormatInfo` after as well.

`GetNumberFormatInfo` is using the json file, but the formatting come from CultureInfo from the OS.

The OS says ARS and COP have 2 decimals, and wikipedia seems to agree, so I am not sure why it should be changed.

**Comment by NicolasDorier** [comment]
Please just add a test on the `GetNumberFormatInfo` https://github.com/btcpayserver/btcpayserver/pull/5710#discussion_r1475590651 in the same test method (below after the loop).

File: BTCPayServer.Tests/FastTests.cs (hunk 5 above)
Comment (TChukwuleta) [reply BTCPayServer.Tests/FastTests.cs:793]: Well according to the issue, https://github.com/btcpayserver/btcpayserver/issues/5708#issue-2102859157, they no longer run cents

**Pull request merged**